pokemon_wordle.py

```diff
@@ -15,207 +19,256 @@ class Pokemon:
     type_02: str = ""
 
 
-def guide():
-    """Display a guide.
-    ゲームのルールを表示する。
-    """
-    print("5文字のポケモンの名前を当てるゲームです！\n")
-    print(cl.Fore.YELLOW + "文字だけ合っていたら黄色で、")
-    print(cl.Fore.GREEN + "文字も位置も合っていたら緑色で、")
-    print(cl.Fore.WHITE + "違っていたら白色の \"・\" で表示します。\n")
+class SystemCommand(Enum):
+    HELP = "help"
+    QUIT = "quit"
+    HINT = "hint"
+
+
+class Color(Enum):
+    GREEN = colorama.Fore.GREEN
+    YELLOW = colorama.Fore.YELLOW
+    WHITE = colorama.Fore.WHITE
+
+
+class ColorLabels:
+    def __init__(self, answer: str, response: str) -> None:
+        self.answer = answer
+        self.response = response
+        self.labels = self._get_color_labels()
+
+    def _get_need_to_label_list(self) -> list[str]:
+        answer = self.answer
+        response = self.response
+        answer_counter = Counter(answer)
+        response_counter = Counter(response)
+        char_set_intersection = set(list(response)) & set(list(answer))
+
+        need_to_label_list: list[str] = []
+        for char in char_set_intersection:
+            for _ in range(min(response_counter[char], answer_counter[char])):
+                need_to_label_list.append(char)
+        return need_to_label_list
+
+    def _get_color_labels(self) -> list[Color]:
+        answer = self.answer
+        response = self.response
+        labels: list[Color] = [Color.WHITE] * 5
+        need_to_label_list = self._get_need_to_label_list()
+
+        for index, (response_char, answer_char) in enumerate(zip(response, answer)):
+            if response_char == answer_char:
+                labels[index] = Color.GREEN
+                need_to_label_list.remove(response_char)
+
+        for index, (response_char, answer_char) in enumerate(zip(response, answer)):
+            if response_char in need_to_label_list:
```

Comment:
65行目、この判定だとすでに GREENが入っていても YELLOWで上書きされちゃうケースがありますね
こうすると解決できそうです
```suggestion
            if labels[index] != Color.GREEN and response_char in need_to_label_list:
```

Reply:
これみたいに
- 同じ文字が複数入っている
- 片方は GREENでもう片方は YELLOW
というときに発生しそうな予感
![スクリーンショット 2022-07-03 13 02 45](https://user-images.githubusercontent.com/18657640/177024086-63153647-0167-4543-a076-4ddf9393edc2.png)

Reply:
ほんとだ、ありがとうございますm(__)m